app/controllers/tweet_controller.rb

```diff
@@ -0,0 +1,44 @@
+class TweetController < ApplicationController
+  get '/tweets' do
+    redirect '/login' unless session.key?(:user_id)
+    @tweets = Tweet.all
+    @username = User.find(session[:user_id]).username
+    erb :'tweets/index'
+  end
+
+  get '/tweets/new' do
+    redirect '/login' unless session.key?(:user_id)
+    erb :'tweets/new'
+  end
+
+  post '/tweets' do
+    redirect '/login' unless session.key?(:user_id)
```

Comment:
A `logged_in?` method defined as a helper in the ApplicationController could be useful here, which is preferable to referring to the session object directly.